saleor/core/permissions.py

```diff
@@ -98,6 +98,43 @@ class SitePermissions(BasePermissionEnum):
 ]
 
 
+def is_app(context):
+    return bool(context.app)
+
+
+def is_user(context):
+    return context.user.is_active and context.user.is_authenticated
+
+
+def is_staff_user(context):
+    return is_user(context) and context.user.is_staff
+
+
+class PermissionFunctions(BasePermissionEnum):
+    # Grants access to any authenticated app.
+    AUTHENTICATED_APP = "permission_functions.authenticated_app"
+
+    # Grants access to any authenticated staff user.
+    AUTHENTICATED_STAFF_USER = "permission_functions.authenticated_staff_user"
+
+    # Grants access to any authenticated user.
+    AUTHENTICATED_USER = "permission_functions.authenticated_user"
+
+    # Grants access to the owner of the related object. This rule doesn't come with any
+    # permission function, as the ownership needs to be defined individually in each
+    # case.
+    OWNER = "permission_functions.owner"
+
+
+def resolve_internal_permission_fn(perm):
+    fn_map = {
```

Comment:
I would not create dict every time I will call this function. Will be better to move this mapping outside.

Reply:
Fixed